src/getty-generator/getty-generator.c

```diff
@@ -190,72 +274,73 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
         if (r < 0)
                 log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
 
-        r = getenv_for_pid(1, "SYSTEMD_GETTY_AUTO", &getty_auto);
-        if (r < 0)
-                log_warning_errno(r, "Failed to parse $SYSTEMD_GETTY_AUTO environment variable, ignoring: %m");
-        else if (r > 0) {
-                r = parse_boolean(getty_auto);
-                if (r < 0)
-                        log_warning_errno(r, "Failed to parse $SYSTEMD_GETTY_AUTO value \"%s\", ignoring: %m", getty_auto);
-                else
-                        arg_enabled = r;
-        }
+        parse_env();
+        parse_credentials();
 
-        if (!arg_enabled) {
+        if (arg_getty_sources == GETTY_SOURCE_NONE) {
                 log_debug("Disabled, exiting.");
                 return 0;
         }
 
-        r = add_credential_gettys();
-        if (r < 0)
-                return r;
+        if (FLAGS_SET(arg_getty_sources, GETTY_SOURCE_CREDENTIAL)) {
+                r = add_credential_gettys();
+                if (r < 0)
+                        return r;
+        }
 
-        if (detect_container() > 0)
+        if (detect_container() > 0) {
```

Comment:
Ah!  I unintentionally changed the intent of this logic in my patch.  Thanks for cleaning this up.